cfme/tests/cloud/test_providers.py

```diff
@@ -94,6 +96,43 @@ def test_discovery_password_mismatch_validation_cloud(appliance):
     view.flash.assert_message('Password/Verify Password do not match')
 
 
+@pytest.mark.tier(3)
+@pytest.mark.uncollectif(lambda: store.current_appliance.version >= '5.9',
+                         reason='no more support for cloud provider discovery')
+def test_discovery_error_azure_cloud(appliance):
+    """ Test Azure discovery with feck data
+
+    prerequisites:
+        * appliance supporting discovery
+
+    Steps:
+        * Navigate Cloud provider discovery and select Azure
+        * Fill all fields with feck data
+        * Start Discovery
+        * Even with wrong data discovery will start with the proper flash message assert it
+        * Check for provider should not discover
+    """
+    cred = Credential(
+        principal=fauxfactory.gen_alphanumeric(5),
+        secret=fauxfactory.gen_alphanumeric(8),
+        tenant_id=fauxfactory.gen_alphanumeric(10),
+        subscription_id=fauxfactory.gen_alphanumeric(10))
+
+    collection = appliance.collections.cloud_providers
+    view = navigate_to(collection, 'All')
+    initial_count = len(view.entities.entity_names)
+
+    collection.discover(cred, AzureProvider)
+    view = appliance.browser.create_view(CloudProvidersView)
+    view.flash.assert_success_message('Cloud Providers: Discovery successfully initiated')
+
+    try:
+        collection.wait_for_new_provider()
+    except TimedOutError:
+        logger.warning('Timed out waiting for new provider, continuing')
```

Comment:
**QUESTION**
What's the point in this exception?

Reply:
It's not necessary but we are handling exception so better to log it.

Reply:
Ok, so why do you wait for a new provider if it's not necessary?

Reply:
```python
with pytest.raises(TimedOutError):
    collection.wait_for_new_provider()
```
looks much better